packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -1495,7 +1495,7 @@ class _NestedScrollPosition extends ScrollPosition implements ScrollActivityDele
       // the whole NestedScrollView, since it is possible to have different
       // ScrollPhysics for the inner and outer positions.
       physics.shouldAcceptUserOffset(this)
-        || innerCanDrag,
+        || innerCanDrag || pixels != 0.0
```

Comment:
This does not look right, why would the pixels affect this? The scroll physics already determine whether or not we can scroll out of bounds.

Reply:
In a NestedScrollView, the entire area can only be scrolled if both the outer and inner parts are unable to scroll. If the outer part is set to NeverScrollableScrollPhysics and the inner part can scroll, then the entire area can be scrolled. During the scrolling process, the updateCanDrag function is called to update whether scrolling is possible.

When the inner part is scrolling, it prioritizes scrolling the outer part. If the maximum scroll size of the inner part is smaller than the size of the NestedScrollView, the inner part is updated to be non-scrollable during the scrolling process. Combined with the fact that the outer part is non-scrollable, the entire NestedScrollView becomes non-scrollable. However, at this point, the outer part has already scrolled partially. This can result in the situation shown in the video where the scrolling gets stuck.

Therefore, when setting it to be non-scrollable, a check is added to determine whether the outer part has already scrolled by examining the pixels. If scrolling has occurred, it cannot be set to non-scrollable.

Reply:
> If the outer part is set to NeverScrollableScrollPhysics and the inner part can scroll, then the entire area can be scrolled. 

The NestedScrollView is intended to compose two scroll views together as one. If they have differing scroll physics, then they will have unique effects like this. I don't think it is safe to assume we should disallow scrolling like this.

> However, at this point, the outer part has already scrolled partially. This can result in the situation shown in the video where the scrolling gets stuck.

I do think this is a bug, but I don't know that this is the way to resolve it. Should we instead check the other scrollable when scrolling back is the other cannot scroll?

Reply:
When this situation occurs, both the inner and outer elements are not allowed to scroll, and we don't have an appropriate opportunity to restore scrolling.
We can only determine in advance whether to enable scrolling based on the specific circumstances. I believe the fundamental reason here is that the outer element is set to not scroll, but due to the characteristics of NestedScrollView, the outer element can still scroll. Therefore, when setting it to not scroll, we should check if the outer element has experienced scrolling to resolve this issue.

Reply:
Right. Thank you for explaining the issue. I just don't think this is the best way to resolve it since it does not take into account the scroll physics. Just relying on the pixels of the position means there will be other issues where this does not work as expected.

Reply:
pixels is an additional or condition added after the scroll physics returns a result.
Only when a freeze occurs will the judgment reach this point; under normal circumstances, it will not be reached.
I think this is the most appropriate solution here (or it is the solution that currently conforms to the design principles of NestedScrollView).

Reply:
Right, so if the physics returns false, the or overrides it. Unfortunately, we cannot make this change. If you would like to propose a different solution, it would be welcome. The current solution here is not one that will appropriately resolve this issue.